lib/registration.pm

```diff
@@ -255,17 +270,17 @@ sub fill_in_registration_data {
 
     # Process modules on sle 15
     if (is_sle '15+') {
-        my $modules_needle = "modules-preselected-" . get_required_var('SLE_PRODUCT');
+        my $modules_needle = ("modules-preselected-" . get_required_var('SLE_PRODUCT'));
         if (check_var('BETA', '1')) {
             assert_screen('scc-beta-filter-checkbox');
             send_key('alt-i');
         }
-        elsif (!check_screen($modules_needle, 0)) {
+        elsif (check_screen('scc-beta-filter-checkbox', 0)) {
             record_info('bsc#1094457 : SLE 15 modules are still in BETA while product enter GMC phase');
-            assert_screen('scc-beta-filter-checkbox');
             send_key('alt-i');
         }
-        assert_screen($modules_needle);
+        verify_preselected_modules unless (check_screen($modules_needle, 0));
```

Comment:
Small concern, I don't see that we validate that modules list was already loaded above. Before we had 30 seconds timeout. Can it be the case that we were to fast and list wasn't shown yet, so `$modules_needle` fails and we fallback to verify_preselected_modules? Also, to be honest I would move that logic to verify_preselected_modules and not use it just as a fallback.